pandas/core/groupby/groupby.py

```diff
@@ -3002,6 +3051,7 @@ def quantile(self, q=0.5, interpolation: str = "linear"):
         a    2.0
         b    3.0
         """
+        numeric_only_bool = self._resolve_numeric_only(numeric_only)
```

Comment:
why don't u use this in the other function (id min and so on)

Reply:
idxmax/idxmin need to default to None for backwards compatibility; using _resolve_numeric_only would currently have them default to True. The different being that currently idxmin/idxmax will operate on categorical columns with numeric_only being None; with numeric_only True they would be dropped.

I do plan on making all necessary methods go through _resolve_numeric_only as part of #46072, including idxmin/idxmax. The logic would there would need to be made method-specific, and was going to do all methods together.